source/Octopus.Server.Client/Model/AuthenticationConfigResource.cs

```diff
@@ -35,5 +35,8 @@ public AuthenticationConfigResource()
 
         [Writeable]
         public bool RememberMeEnabled { get; set; }
+
+        [Writeable]
+        public bool ApiKeysEnabledForNonServiceAccounts { get; set; }
```

Comment:
Imagine we had 3 types of accounts: Apple, Blueberry and Watermelon.
The first requirement is to disable api keys for all but Watermelon.
```
ApiKeysEnabledForNonWatermelonAccounts
```

How would the code need to change in the following scenarios:
- The concept of Watermelon accounts is removed.
- A new requirement is added for finer-grained control: Apple or Blueberry accounts can be disabled.

Reply:
Are you suggesting to use something like `ForUserAccounts` instead of `ForNonXAccounts`? It would make it easier to change. Originally I went with NonService since I thought "User" could be a bit confusing. Service accounts are technically users as well. Not sure if there's a more descriptive name for "user accounts".